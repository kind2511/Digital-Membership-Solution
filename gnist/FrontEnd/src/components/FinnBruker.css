.finn-bruker-container {
  background-color: #f5f5f5;
  border: 1px solid #ccc;
  border-radius: 8px;
  padding: 20px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.finn-bruker-title {
  color: #333;
  font-size: 20px;
  margin-bottom: 10px;
}

.finn-bruker-input {
  width: 95%;
  padding: 10px;
  margin-bottom: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
}

.finn-bruker-status {
  color: #666;
  margin-bottom: 10px;
}

.finn-bruker-results {
  background-color: #fff;
  width: 95%;
  padding: 10px;
  border-radius: 4px;
  max-height: 300px;
  overflow-y: auto;
}

.finn-bruker-result {
  padding: 10px;
  border-bottom: 1px solid #eee;
  cursor: pointer;
}

.finn-bruker-result:hover {
  background-color: #f7f7f7;
}

.finn-bruker-result:last-child {
  border-bottom: none;
}

.user-details {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: #fff;
  padding: 20px;
  border-radius: 8px;
  width: 80%;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  z-index: 100;
}

.user-details h3 {
  color: #333;
  font-size: 18px;
  margin-bottom: 10px;
}

.user-details p {
  margin-bottom: 8px;
}

.user-details p:last-child {
  margin-bottom: 0;
}

.close-details-button {
  padding: 8px 16px;
  background-color: #d9534f;
  color: white;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

.close-details-button:hover {
  background-color: #c9302c;
}
