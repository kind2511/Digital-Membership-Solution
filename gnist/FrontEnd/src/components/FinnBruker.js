import React, { useState } from 'react';
import './FinnBruker.css';

function FinnBruker() {
    const [searchTerm, setSearchTerm] = useState('');
    const [results, setResults] = useState([]);
    const [searchStatus, setSearchStatus] = useState('');
    const [selectedUser, setSelectedUser] = useState(null);

    const fetchData = (value) => {
        if (value.trim() === '') {
            setResults([]);
            setSearchStatus('');
            return;
        }
        setSearchStatus('Searching...');
        fetch(`http://127.0.0.1:8000/digital_medlemsordning/search_member/?name=${value}`)
            .then((response) => {
                if (!response.ok) {
                    throw new Error(`HTTP error! status: ${response.status}`);
                }
                return response.json();
            })
            .then((data) => {
                setResults(data);
                setSearchStatus(`Found ${data.length} results.`);
            })
            .catch((error) => {
                console.error("Error fetching data:", error);
                setSearchStatus("Failed to fetch data. Please try again later.");
            });
    };

    const handleSearchChange = (e) => {
        const value = e.target.value;
        setSearchTerm(value);
        fetchData(value);
    };

    const handleUserSelect = (user) => {
        setSelectedUser(user);
    };

    const handleCloseDetails = () => {
        setSelectedUser(null);
    };

    return (
        <div className="finn-bruker-container">
            <h2 className="finn-bruker-title">Finn Bruker</h2>
            <input
                type="text"
                className="finn-bruker-input"
                placeholder="Søk etter medlem..."
                value={searchTerm}
                onChange={handleSearchChange}
            />
            <div className="finn-bruker-status">{searchStatus}</div>
            <div className="finn-bruker-results">
                {results.map((user, index) => (
                    <div key={index} className="finn-bruker-result" onClick={() => handleUserSelect(user)}>
                        {`${user.first_name} ${user.last_name}`}
                    </div>
                ))}
            </div>
            {selectedUser && (
                <div className="user-details">
                    <h3>Detaljer:</h3>
                    <p>Navn: {selectedUser.first_name} {selectedUser.last_name}</p>
                    <p>Fødselsdato: {selectedUser.birthdate || 'N/A'}</p>
                    <p>Kjønn: {selectedUser.gender}</p>
                    <p>E-post: {selectedUser.email}</p>
                    <p>Telefon: {selectedUser.phone_number}</p>
                    <p>Verge Navn: {selectedUser.guardian_name || 'N/A'}</p>
                    <p>Verge Telefon: {selectedUser.guardian_phone || 'N/A'}</p>
                    <p>Verifisert: {selectedUser.verified ? 'Ja' : 'Nei'}</p>
                    <p>Status : {selectedUser.banned ? 'Rød' : 'Grønn'}</p>
                    <p>Status Dato: {selectedUser.banned_from || 'N/A'} til {selectedUser.banned_until || 'N/A'}</p>
                    <p>Ekstra Info: {selectedUser.info || 'N/A'}</p>
                    <button className="close-details-button" onClick={handleCloseDetails}>Lukk</button>
                </div>
            )}
        </div>
    );
}

export default FinnBruker;
