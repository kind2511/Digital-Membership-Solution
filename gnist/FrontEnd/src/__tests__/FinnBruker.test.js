import React from 'react';
import renderer from 'react-test-renderer';
import FinnBruker from '../components/FinnBruker';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Mock axios for the test
const mock = new MockAdapter(axios);
const usersMock = [
  {
    auth0ID: 'auth0|123456789',
    first_name: 'John',
    last_name: 'Doe',
    birthdate: '2000-01-01',
    gender: 'Male',
    email: 'john.doe@example.com',
    phone_number: '12345678',
    guardian_name: 'Jane Doe',
    guardian_phone: '87654321',
    verified: true,
    banned: false,
    banned_from: null,
    banned_until: null,
    info: 'Some extra info about John Doe'
  },
  {
    auth0ID: 'auth0|987654321',
    first_name: 'Jane',
    last_name: 'Doe',
    birthdate: '2002-02-02',
    gender: 'Female',
    email: 'jane.doe@example.com',
    phone_number: '87654321',
    guardian_name: null,
    guardian_phone: null,
    verified: false,
    banned: true,
    banned_from: '2023-01-01',
    banned_until: '2023-12-31',
    info: 'Some extra info about Jane Doe'
  }
];
mock.onGet('http://127.0.0.1:8000/digital_medlemsordning/search_member/?name=').reply(200, usersMock);

test('FinnBruker component snapshot', () => {
  const component = renderer.create(<FinnBruker />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
