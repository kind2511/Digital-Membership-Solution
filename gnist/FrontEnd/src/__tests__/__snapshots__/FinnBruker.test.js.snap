// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinnBruker component snapshot 1`] = `
<div
  className="finn-bruker-container"
>
  <h2
    className="finn-bruker-title"
  >
    Finn Bruker
  </h2>
  <input
    className="finn-bruker-input"
    onChange={[Function]}
    placeholder="SÃ¸k etter medlem..."
    type="text"
    value=""
  />
  <div
    className="finn-bruker-status"
  />
  <div
    className="finn-bruker-results"
  />
</div>
`;
